From d3fd5365fb9fad89af47e17da35cef1a34aff1d3 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Tue, 19 Apr 2016 15:07:18 +0200
Subject: [PATCH] IPA: terminate properly if view name lookup fails

Since commit 5a5f1e1053415efaa99bb4d5bc7ce7ac0a95b757 the view name
lookup is the last step in the subdomain lookup request. In case of an
error the request should be finished and no previous step should be
called again.

Resolves https://fedorahosted.org/sssd/ticket/2993

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
(cherry picked from commit 57d8b4b9254442a568838cb60ea16068965f2df0)
---
 src/providers/ipa/ipa_subdomains.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/src/providers/ipa/ipa_subdomains.c b/src/providers/ipa/ipa_subdomains.c
index cb5a23bfb8043e620061e11d5e567d3e39eab6e3..62796b8d15682aedb952b9c46561867d82684b80 100644
--- a/src/providers/ipa/ipa_subdomains.c
+++ b/src/providers/ipa/ipa_subdomains.c
@@ -840,13 +840,7 @@ static void ipa_get_view_name_done(struct tevent_req *req)
         if (ret == EOPNOTSUPP || ret == EIO) {
             DEBUG(SSSDBG_TRACE_FUNC, "get_view_name request failed, looks " \
                                      "like server does not support views.\n");
-            ret = ipa_check_master(ctx);
-            if (ret == EAGAIN) {
-                return;
-            } else if (ret != EOK) {
-                goto done;
-            }
-
+            ret = EOK;
         } else {
             DEBUG(SSSDBG_OP_FAILURE, "get_view_name request failed.\n");
         }
-- 
2.7.3

